### PR TITLE
OneBranch Build

### PR DESCRIPTION
**Pull request by littleaj**
* Adds OneBranch .pipelines folder with build definitions
* Adds .scripts folder with build helper scripts (installs dependencies, and other env setup, if needed in the future).
* Adds dependency lockfiles because dependency scan is done in parallel w/o an opportunity to generate on the fly.

### CODE REVIEW

File: buildSrc/src/main/groovy/com/microsoft/applicationinsights/build/tasks/PropertiesFileGenerator.groovy [hunk 1]
```diff
@@ -44,8 +44,8 @@ class PropsFileGen extends DefaultTask {
         }
 
         Properties oldProps = new Properties();
-        targetFile.withInputStream { ifs ->
```
Comment (littleaj) [inline buildSrc/src/main/groovy/com/microsoft/applicationinsights/build/tasks/PropertiesFileGenerator.groovy:47]: fix file stream closure

**Comment by heyams** [comment]
@trask smoke tests are finally green 🥰

**Comment by heyams** [comment]
smoke test passed locally.. it must be github action?  lockfiles were updated. @trask

**Comment by trask** [comment]
@heyams I updated the lock files, smoke tests passing now, I think you need to publishToMavenLocal latest from otel submodule

**Review by trask — APPROVED** [review]
thx @littleaj @heyams!

a couple of questions below, but going to go ahead and merge b/c I want to merge #1482, but want this to go in first

File: .pipelines/pipeline.user.windows.buddy.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          exclude:
+            - '**/*.pdb'
+          signing_options:
+            profile: external_distribution
+            sign_inline: true
+      logs:
+        - to: 'PerfCounters DLL Build Logs'
+          from: 'core/build/tmp'
+          include:
+            - '*WindowsCpp/*'
```
[the remark this answers is not in the document]
Comment (heyams) [reply .pipelines/pipeline.user.windows.buddy.yml:26]: I think Windows 10 SDK takes care of etw.

File: .pipelines/pipeline.user.windows.buddy.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    - !!buildcommand
+      name: 'Assemble ApplicationInsights-Java JARs'
+      command: '.scripts/gradle.cmd'
```
[the remark this answers is not in the document]
Comment (heyams) [reply .pipelines/pipeline.user.windows.buddy.yml:29]: we can.

File: .pipelines/pipeline.user.windows.yml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  commands:
+    - !!defaultcommand
+      name: 'Generate Version'
+      command: '.scripts/version.cmd'
+signing_options:
+  profile: '170'
+static_analysis_options:
+  policheck_options:
+    disable_tool_scan: false
+    fail_on_error: true
```
[the remark this answers is not in the document]
Comment (heyams) [reply .pipelines/pipeline.user.windows.yml:20]: here is the `signing_options`: https://onebranch.visualstudio.com/Pipeline/_wiki/wikis/Pipeline.wiki/316/Signing?anchor=enabling-signing.  I think we can search `versioning` & `static_analysis_options` there.

**Pull request merged**